### DISCUSSION TITLE
How to sum a BIGINT column

### DISCUSSION THREAD

**Discussion by benthecarman** [diesel-rs/diesel#4000] [Q&A] [answered]
I am trying to do a `SUM` on a `BIGINT` column but it doesn't seem possible. Is there any examples on how to do this? If I change my column to an i32 the normal examples work, not sure why it doesn't work for i64

**Comment by weiznich** [comment]
There is no fundamental difference between summing a `INTEGER` and a `BIGINT` column in diesel. You might want to check if you want to check if you specified the right return type on rust side. It needs to correspond to [this type for sums of `BigInt`'s](https://docs.diesel.rs/2.1.x/diesel/sql_types/trait.Foldable.html#associatedtype.Sum-1) (as it is specified by the sql standard).

Otherwise you need to provide a minimal example, including the complete error message you get from rustc if you need further help.

**Comment by benthecarman** [comment]
This is what I was doing

```rust
 let sum: Option<i64> = my_table::table
                .select(diesel::dsl::sum(my_table::big_int_col))
                .first(conn)?;
```

and I get this:

```
the trait bound `i64: FromSql<diesel::sql_types::Numeric, Pg>` is not satisfied                      
     --> src/models/routed_payment.rs:73:24                                                                            
      |                                                                                                                
 73   |                 .first(conn)?;                                                                                 
      |                  ----- ^^^^ the trait `FromSql<diesel::sql_types::Numeric, Pg>` is not implemented for `i64`, which is required by `SelectStatement<FromClause<my_table::table>
```

**Reply by weiznich** [reply, marked as answer]
Well as already written above: You need to use a rust side type that is compatible with the `Numeric` sql type. That's not the case for `i64` so that's expected behavior. See the [documentation](https://docs.diesel.rs/2.1.x/diesel/sql_types/struct.Numeric.html) for a list of compatible types.
